common/build/eslint-plugin-fluid/src/test/enforce-no-unchecked-record-access/enforce-no-unchecked-record-access.test.js

```diff
@@ -0,0 +1,67 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+const assert = require("assert");
+const path = require("path");
+const { ESLint } = require("eslint");
+
+describe("ESLint Rule Tests", function () {
+	async function lintFile(file) {
+		const eslint = new ESLint({
+			useEslintrc: false,
+			overrideConfig: {
+				rules: {
+					"no-unchecked-record-access": "error",
+				},
+				parser: "@typescript-eslint/parser",
+				parserOptions: {
+					project: path.join(__dirname, "../tsconfig.json"),
+				},
+			},
+			rulePaths: [path.join(__dirname, "../../rules")],
+		});
+		const fileToLint = path.join(__dirname, "../mockFiles/no-unchecked-record-access", file);
+		const results = await eslint.lintFiles([fileToLint]);
+		return results[0];
+	}
+
+	it.only("Should report an error for unchecked record access", async function () {
+		const result = await lintFile("fileWithOnlyRecordAccess.ts");
+		assert.strictEqual(result.errorCount, 7, "Should have 7 errors");
+		assert.strictEqual(
+			result.messages[0].message,
+			"'nestedObj.nested.a' is possibly 'undefined'",
+		);
```

Comment:
Is it possible to make these checks more specific? These seem like they would pass if the defect was detected on a different line than where expected.